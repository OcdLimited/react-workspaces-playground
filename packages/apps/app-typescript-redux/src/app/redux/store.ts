import { configureStore, ThunkAction, Action, StoreEnhancer } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './rootReducer';
import rootSaga from './sagas/sagas';

const enhancers = new Array<StoreEnhancer>();

if (process.env.NODE_ENV !== 'production') {
	enhancers.push(withReduxEnhancer);
}

const sagaMiddleware = createSagaMiddleware();

export const store = {
	...configureStore({
		reducer: rootReducer,
		enhancers: enhancers,
		middleware: [sagaMiddleware],
	}),
	injectedSagas: [],
	runSaga: sagaMiddleware.run,
};

sagaMiddleware.run(rootSaga);

if (process.env.NODE_ENV === 'development' && module.hot) {
	module.hot.accept('./rootReducer', () => {
		const newRootReducer = require('./rootReducer').default;
		store.replaceReducer(newRootReducer);
	});
}

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
export type AppDispatch = typeof store.dispatch;

export default store;
