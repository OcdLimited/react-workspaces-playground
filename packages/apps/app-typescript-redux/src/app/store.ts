import { configureStore, ThunkAction, Action, StoreEnhancer } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import counterReducer from '../features/counter/counterSlice';

const enhancers = new Array<StoreEnhancer>();

if (process.env.NODE_ENV !== 'production') {
  enhancers.push(withReduxEnhancer);
}

export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
  enhancers: enhancers,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
