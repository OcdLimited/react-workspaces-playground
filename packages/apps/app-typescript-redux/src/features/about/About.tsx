import React from 'react';
import logo from './logo.svg';
import { Counter } from '../counter/Counter';

import './About.css';

export function About() {
	return (
		<div className="App">
			<header className="App-header">
				<img src={logo} className="App-logo" alt="logo" />
				<Counter />
				<p>ocd.codes react sdk</p>
				<span>
					<span>Learn </span>
					<a className="App-link" href="https://reactjs.org/" target="_blank" rel="noopener noreferrer">
						React
					</a>
					<span>, </span>
					<a className="App-link" href="https://react.i18next.com" target="_blank" rel="noopener noreferrer">
						react-i18next
					</a>
					<span>, </span>
					<a className="App-link" href="https://material-ui.com/" target="_blank" rel="noopener noreferrer">
						Material UI
					</a>
					<span>, </span>
					<a className="App-link" href="https://jaredpalmer.com/formik/" target="_blank" rel="noopener noreferrer">
						Formik
					</a>
					<span>, </span>
					<a className="App-link" href="https://redux.js.org/" target="_blank" rel="noopener noreferrer">
						Redux
					</a>
					<span>, </span>
					<a className="App-link" href="https://redux-toolkit.js.org/" target="_blank" rel="noopener noreferrer">
						Redux Toolkit
					</a>
					<span>, </span>
					<a className="App-link" href="https://react-redux.js.org/" target="_blank" rel="noopener noreferrer">
						React Redux
					</a>
					,<span> and </span>
					<a className="App-link" href="https://redux-saga.js.org" target="_blank" rel="noopener noreferrer">
						Redux Sagas
					</a>
				</span>
			</header>
		</div>
	);
}

export default About;
