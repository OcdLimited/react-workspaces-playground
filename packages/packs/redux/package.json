{
	"name": "@ocdlimited/abp.react.redux",
	"version": "0.0.4-alpha.0",
	"main": "src/index.ts",
	"main:src": "src/index.ts",
	"scripts": {
		"test": "react-scripts test",
		"lint": "eslint ./src/**/*.ts ./src/*.ts --max-warnings=0 --format=codeframe"
	},
	"dependencies": {
		"@ocdlimited/react-scripts": "^0.0.4-alpha.0",
		"@reduxjs/toolkit": "^1.2.5",
		"@testing-library/jest-dom": "^4.2.4",
		"@testing-library/react": "^9.4.0",
		"@testing-library/user-event": "^8.0.3",
		"@types/jest": "^24.0.25",
		"@types/node": "^13.1.6",
		"@types/react": "^16.9.17",
		"@types/react-dom": "^16.9.4",
		"@types/react-redux": "^7.1.7",
		"@typescript-eslint/eslint-plugin": "^2.15.0",
		"@typescript-eslint/parser": "^2.15.0",
		"addon-redux": "^1.1.0",
		"react": "^16.13.1",
		"react-dom": "^16.13.1",
		"react-redux": "^7.2.0",
		"react-router": "6.0.0-alpha.4",
		"react-router-dom": "6.0.0-alpha.4",
		"redux-saga": "1.1.3",
		"typescript": "~3.8.2"
	},
	"eslintConfig": {
		"extends": "react-app"
	},
	"browserslist": [
		">0.2%",
		"not dead",
		"not ie <= 11",
		"not op_mini all"
	],
	"jest": {
		"collectCoverageFrom": [
			"src/**/*.{ts,tsx}",
			"!src/setupTests.{ts,tsx}",
			"!src/serviceWorker.{ts,jsx}",
			"!src/**/*.(spec|test|stories).{ts,tsx}"
		],
		"coverageThreshold": {
			"global": {
				"statements": 95,
				"branches": 80,
				"functions": 90,
				"lines": 98
			}
		}
	},
	"license": "MIT"
}
