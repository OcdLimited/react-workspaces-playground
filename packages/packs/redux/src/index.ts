export * from './sagas';
export * from './reducers';
export * from './store';
export * from './types';
