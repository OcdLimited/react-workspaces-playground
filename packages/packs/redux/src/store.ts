import { configureStore, StoreEnhancer, Store, AnyAction } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import createSagaMiddleware from 'redux-saga';
import { createApiMiddleware, appConfigSaga } from '@ocdlimited/abp.react.core';
import { createRootReducer } from './reducers/rootReducer';

export interface InjectableStore extends Store<any, AnyAction> {
	injectedReducers: any;
	injectedSagas: any;
	runSaga(saga: any, args: any): any;
}

const enhancers: StoreEnhancer[] = [];

if (process.env.NODE_ENV !== 'production') {
	enhancers.push(withReduxEnhancer);
}

export function buildStore(rootReducer: any, preloadedState?: any): InjectableStore {
	const sagaMiddleware = createSagaMiddleware();

	const store = {
		...configureStore({
			reducer: createRootReducer(rootReducer),
			enhancers: enhancers,
			middleware: [createApiMiddleware(), sagaMiddleware],
			preloadedState,
		}),
		injectedSagas: [],
		injectedReducers: { ...rootReducer },
		runSaga: sagaMiddleware.run,
	};

	sagaMiddleware.run(appConfigSaga);

	/* istanbul ignore next */
	if (process.env.NODE_ENV === 'development' && module.hot) {
		if (module.hot) {
			// Enable Webpack hot module replacement for reducers
			module.hot.accept('./reducers', () => {
				const nextReducer = createRootReducer(store.injectedReducers);
				store.replaceReducer(nextReducer);
			});
		}
	}

	return store;
}
