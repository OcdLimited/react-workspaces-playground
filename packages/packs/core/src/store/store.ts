import { combineReducers } from '@reduxjs/toolkit';
import { InjectableStore, buildStore as innerBuildStore } from '@ocdlimited/abp.react.redux';
import { apiReducer, createApiMiddleware } from '../api';
import { appConfigReducer, appConfigSaga } from '../application-configuration';
import { toastReducer } from '../toast';
import { tokenReducer } from '../token';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function createRootReducer(injectedReducers: any): any {
	return combineReducers({
		api: apiReducer,
		config: appConfigReducer,
		toast: toastReducer,
		token: tokenReducer,
		...injectedReducers,
	});
}

export function buildStore(rootReducer: unknown, preloadedState?: unknown): InjectableStore {
	const reducer = createRootReducer(rootReducer);
	return innerBuildStore(reducer, createRootReducer, preloadedState, [createApiMiddleware()], [appConfigSaga]);
}

export default buildStore;
