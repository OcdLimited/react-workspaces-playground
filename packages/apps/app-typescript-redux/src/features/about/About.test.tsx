import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../app/redux/store';
import About from './About';

it('renders learn react link', () => {
	const { container, getByText } = render(
		<Provider store={store}>
			<MemoryRouter>
				<About />
			</MemoryRouter>
		</Provider>,
	);

	expect(getByText(/learn/i)).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
