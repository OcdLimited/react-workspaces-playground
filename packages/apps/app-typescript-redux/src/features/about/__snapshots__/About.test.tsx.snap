// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <div>
        <div
          class="row"
        >
          <button
            aria-label="Increment value"
            class="button"
          >
            +
          </button>
          <span
            class="value"
          >
            0
          </span>
          <button
            aria-label="Decrement value"
            class="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <input
            aria-label="Set increment amount"
            class="textbox"
            value="2"
          />
          <button
            class="button"
          >
            Add Amount
          </button>
          <button
            class="asyncButton"
          >
            Add Async
          </button>
        </div>
      </div>
      <p>
        <a
          href="/account/login"
        >
          Login
        </a>
      </p>
      <p>
        ocd.codes react sdk
      </p>
      <span>
        <span>
          Learn 
        </span>
        <a
          class="App-link"
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          React
        </a>
        <span>
          , 
        </span>
        <a
          class="App-link"
          href="https://react.i18next.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          react-i18next
        </a>
        <span>
          , 
        </span>
        <a
          class="App-link"
          href="https://material-ui.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Material UI
        </a>
        <span>
          , 
        </span>
        <a
          class="App-link"
          href="https://jaredpalmer.com/formik/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Formik
        </a>
        <span>
          , 
        </span>
        <a
          class="App-link"
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Redux
        </a>
        <span>
          , 
        </span>
        <a
          class="App-link"
          href="https://redux-toolkit.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Redux Toolkit
        </a>
        <span>
          , 
        </span>
        <a
          class="App-link"
          href="https://react-redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          React Redux
        </a>
        <span>
           and 
        </span>
        <a
          class="App-link"
          href="https://redux-saga.js.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Redux Sagas
        </a>
      </span>
    </header>
  </div>
</div>
`;
